plugins/modules/foreman_hostgroup.py

```diff
@@ -0,0 +1,216 @@
+    default: None
+  medium:
+    description: Medium name
+    required: False
+    default: None
+  operatingsystem:
+    description: Operatingsystem name
+    required: False
+    default: None
+  parameters:
+    description: List of parameters and values
+    required: false
+    default: None
+  partition_table:
+    description: Partition table name
+    required: False
+    default: None
+  pxe_loader:
+    description: PXE Loader
+    required: False
+    default: None
+  realm:
+    description: Realm name
+    required: false
+    default: None
+  root_pass:
+    description: root password
+    required: false
+    default: None
+  force_update:
+    description: forces update of hostgroup, usefull when in need of password update
+    required: false
+    default: false
+  smart_proxy:
+    description: Smart Proxy name
+    required: False
+    default: None
+  subnet:
+    description: Subnet name
+    required: False
+    default: None
+  server_url:
+    description: foreman url
+    required: true
+  username:
+    description: foreman username
+    required: true
+  password:
+    description: foreman user password
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: verify ssl connection when communicating with foreman
+    default: true
+    type: bool
+  state:
+    description: domain presence
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: "Create a Hostgroup"
+  foreman_hostgroup:
+    name: "new_hostgroup"
+    architecture: "architecture_name"
+    operatingsystem: "operatingsystem_name"
+    medium: "media_name"
+    ptable: "Partition_table_name"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
+
+- name: "Update a Hostgroup"
+  foreman_hostgroup:
+    name: "new_hostgroup"
+    architecture: "updated_architecture_name"
+    operatingsystem: "updated_operatingsystem_name"
+    medium: "updated_media_name"
+    ptable: "updated_Partition_table_name"
+    root_pass: "password"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
+
+- name: "Delete a Hostgroup"
+  foreman_hostgroup:
+    name: "new_hostgroup"
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityApypieAnsibleModule
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'description': 'description',
+    'architecture_id': 'architecture_id',
+    'operatingsystem_id': 'operatingsystem_id',
+    'medium_id': 'medium_id',
+    'ptable_id': 'ptable_id',
+    'root_pass': 'root_pass',
+}
+
+
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
```

Comment:
this spec seems to miss quite a few params you had defined in the documentation. I think it's totally fine to start with just those that you have defined here, but then please adjust the docs :)